src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -1069,5 +1074,27 @@ public UUID getSnapshotBillingProfileId() {
     public CloudPlatform getCloudPlatform() {
       return this.cloudPlatform;
     }
+
+    @Override
+    public String toString() {
```

Comment:
Instead of a custom `toString()`, how about converting this class to a `record`? It looks like all the fields are read-only already.
```
  record SnapshotCacheResult(
      UUID id,
      String name,
      boolean isSelfHosted,
      boolean globalFileIds,
      BillingProfileModel datasetBillingProfileModel,
      UUID snapshotBillingProfileId,
      CloudPlatform cloudPlatform,
      String googleProjectId,
      String datasetProjectId,
      UUID datasetId,
      String datasetName) {

    public SnapshotCacheResult(Snapshot snapshot) {
      this(
          snapshot.getId(),
          snapshot.getName(),
          snapshot.isSelfHosted(),
          snapshot.hasGlobalFileIds(),
          snapshot.getSourceDataset().getDatasetSummary().getDefaultBillingProfile(),
          snapshot.getProfileId(),
          snapshot.getCloudPlatform(),
          Optional.ofNullable(snapshot.getProjectResource())
              .map(GoogleProjectResource::getGoogleProjectId)
              .orElse(null),
          Optional.ofNullable(snapshot.getSourceDataset().getProjectResource())
              .map(GoogleProjectResource::getGoogleProjectId)
              .orElse(null),
          snapshot.getSourceDataset().getId(),
          snapshot.getSourceDataset().getName());
    }
```

Reply:
Great idea!  Working on that now.